### PR TITLE
Fix 557 - crash on illegal USFM sequence.

### PR DESCRIPTION
**Pull request by PhotoNomad0**
Changes:
- parseUsfm - added escaping of illegal USFM sequences and then modified '\\\\' to print out as '\\'  
- Added unit test with illegal USFM sequences.

### CODE REVIEW

File: usfm_tools/support/parseUsfm.py [hunk 1]
```diff
@@ -336,7 +338,14 @@ def clean(unicodeString):
...
     # we work out what to do, non breaking spaces will be ignored
     # ie 0xa0
-    return unicodeString.replace(u'\xa0', ' ')
+    ret_value = unicodeString.replace(u'\xa0', ' ')
+
+    # escape illegal USFM sequences
+    ret_value = ret_value.replace(u'\\ ',  "\\\\ ")
+    ret_value = ret_value.replace(u'\\\n', "\\\\\n")
+    ret_value = ret_value.replace(u'\\\r', "\\\\\r")
+    ret_value = ret_value.replace(u'\\\t', "\\\\\t")
```
Comment (phillip-hopper) [inline usfm_tools/support/parseUsfm.py]: This file is importing `unicode_literals` so all strings are unicode, no need for `u'`.

Comment (phillip-hopper) [reply usfm_tools/support/parseUsfm.py]: Also, mixing single and double quotes.

**Review by phillip-hopper — CHANGES_REQUESTED** [review]

File: usfm_tools/support/parseUsfm.py (hunk 1 above)
Comment (PhotoNomad0) [reply usfm_tools/support/parseUsfm.py]: Thanks, will fix.

**Review by richmahn — CHANGES_REQUESTED** [review]
Find this still fails if the \ is at the very last line of the USFM file. Here are the results of my tests:

```
$ git checkout photonomad0/Fix_557/UsfmConversionFails
HEAD is now at 782b57f... Added escape of \ followed by tab and added nobreak space (\xa0) to test text.  Cleanup of text literals

$ cd usfm_tools/support/

$ python
Python 2.7.13 (v2.7.13:a06454b1afa1, Dec 17 2016, 20:53:40) [MSC v.1500 64 bit (AMD64)] on win32
Type "help", "copyright", "credits" or "license" for more information.
>>> import parseUsfm
>>> parseUsfm.parseString('this is a test \\\nthis is a new line')
[<parseUsfm.TEXTToken object at 0x0000000004508550>, <parseUsfm.EscapedToken object at 0x0000000004508C50>, <parseUsfm.TEXTToken object at 0x0000000004508390>]
>>> parseUsfm.parseString('this is a test \\\n')
[<parseUsfm.TEXTToken object at 0x00000000044FFFD0>, <parseUsfm.EscapedToken object at 0x00000000044FFE10>]
>>> parseUsfm.parseString('this is a test \\')
Expected end of text (at char 15), (line:1, col:16)
'this is a test \\'

$
```

Thus exited on the last command. Is there a way to also let it pass if the last character is a `\`?

**Comment by PhotoNomad0** [comment]
@richmahn - thanks for catching that.  Added that to test case and cleanup.

**Review by phillip-hopper — APPROVED** [review]
LGTM 👍

**Review by richmahn — APPROVED** [review]
LGTM

**Pull request merged**